include/eve/concept/behave_as.hpp

```diff
@@ -0,0 +1,26 @@
+//==================================================================================================
+/*
+  EVE - Expressive Vector Engine
+  Copyright : EVE Project Contributors
+  SPDX-License-Identifier: BSL-1.0
+*/
+//==================================================================================================
+#pragma once
+
+#include <eve/traits/underlying_type.hpp>
+#include <eve/concept/vectorizable.hpp>
+
+namespace eve
+{
+  template<typename T, template<typename> typename Op>
+  concept behave_as = Op<translate_underlying_type_t<T>>::value;
+
+  template<typename T>
+  concept behave_as_plain = plain_scalar_value<translate_underlying_type_t<T>>;
+
+  template<typename T>
+  concept behave_as_integral = integral_scalar_value<translate_underlying_type_t<T>>;
+
+  template<typename T>
+  concept behave_as_floating = floating_scalar_value<translate_underlying_type_t<T>>;
+}
```

Comment:
two questions:

1. Why doesn't `constant` callable framework do the translaction
2. I don't like the `behave` thingy. will write in the chat.

Reply:
I have simplified the impelmetnationa fetr our discussion as I fixed the external type in the external library.